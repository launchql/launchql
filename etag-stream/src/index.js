const stream = require('stream');
const etag = require('etag-hash');
class ETagStream extends stream.Transform {
  constructor(opts = {}) {
    const { partSizeInMb = 5, mode = 'through' } = opts;
    super();
    this.mode = mode;
    this.hash = etag.createHash(partSizeInMb);
  }
  _write(chunk, enc, next) {
    this.hash.update(chunk);
    if (this.mode === 'through') {
      this.push(chunk);
    }
    next();
  }
  _flush(done) {
    if (this.mode === 'through') {
      this.emit('etag', this.hash.digest());
    } else {
      this.push(this.hash.digest());
    }
    done();
  }
}

module.exports = ETagStream;
