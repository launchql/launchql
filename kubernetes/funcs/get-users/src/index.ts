import express, { Request, Response } from 'express';

const app = express();
const port = process.env.PORT || 8080;

app.use(express.json());

interface User {
  id: number;
  name: string;
  email: string;
}

const users: User[] = [
  { id: 1, name: 'John Doe', email: 'john@example.com' },
  { id: 2, name: 'Jane Smith', email: 'jane@example.com' },
  { id: 3, name: 'Bob Johnson', email: 'bob@example.com' }
];

app.get('/', (req: Request, res: Response) => {
  res.status(200).send('Function is running');
});

app.get('/users', (req: Request, res: Response) => {
  res.status(200).json(users);
});

app.post('/users', (req: Request, res: Response) => {
  try {
    const { filter } = req.body as { filter?: string };
    
    if (filter) {
      const filteredUsers = users.filter(user => 
        user.name.toLowerCase().includes(filter.toLowerCase()) || 
        user.email.toLowerCase().includes(filter.toLowerCase())
      );
      
      return res.status(200).json({
        success: true,
        data: filteredUsers,
        count: filteredUsers.length,
        timestamp: new Date().toISOString()
      });
    }
    
    return res.status(200).json({
      success: true,
      data: users,
      count: users.length,
      timestamp: new Date().toISOString()
    });
  } catch (error) {
    console.error('Error processing request:', error);
    return res.status(500).json({
      success: false,
      error: 'Internal server error',
      timestamp: new Date().toISOString()
    });
  }
});

app.listen(port, () => {
  console.log(`Server running on port ${port}`);
});
