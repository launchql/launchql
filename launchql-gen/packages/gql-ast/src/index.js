export const document = ({ definitions }) => ({
  kind: 'Document',
  definitions
});

export const operationDefinition = ({
  operation,
  name,
  variableDefinitions = [],
  directives = [],
  selectionSet
}) => ({
  kind: 'OperationDefinition',
  operation,
  name: {
    kind: 'Name',
    value: name
  },
  variableDefinitions,
  directives,
  selectionSet
});

export const variableDefinition = ({ variable, type, directives }) => ({
  kind: 'VariableDefinition',
  variable,
  type,
  directives
});

export const selectionSet = ({ selections }) => ({
  kind: 'SelectionSet',
  selections
});

export const listType = ({ type }) => ({
  kind: 'ListType',
  type
});

export const nonNullType = ({ type }) => ({
  kind: 'NonNullType',
  type
});

export const namedType = ({ type }) => ({
  kind: 'NamedType',
  name: {
    kind: 'Name',
    value: type
  }
});

export const variable = ({ name }) => ({
  kind: 'Variable',
  name: {
    kind: 'Name',
    value: name
  }
});

export const objectValue = ({ fields }) => ({
  kind: 'ObjectValue',
  fields
});

export const stringValue = ({ value }) => ({
  kind: 'StringValue',
  value
});

export const intValue = ({ value }) => ({
  kind: 'IntValue',
  value
});

export const booleanValue = ({ value }) => ({
  kind: 'BooleanValue',
  value
});

export const listValue = ({ values }) => ({
  kind: 'ListValue',
  values
});

export const nullValue = () => ({
  kind: 'NullValue'
});

export const fragmentDefinition = ({
  name,
  typeCondition,
  directives,
  selectionSet
}) => ({
  kind: 'FragmentDefinition',
  name: {
    kind: 'Name',
    value: name
  },
  typeCondition,
  directives,
  selectionSet
});

export const objectField = ({ name, value }) => ({
  kind: 'ObjectField',
  name: {
    kind: 'Name',
    value: name
  },
  value
});

export const field = ({ name, args, directives, selectionSet }) => ({
  kind: 'Field',
  name: {
    kind: 'Name',
    value: name
  },
  arguments: args,
  directives,
  selectionSet
});

export const argument = ({ name, value }) => ({
  kind: 'Argument',
  name: {
    kind: 'Name',
    value: name
  },
  value
});
