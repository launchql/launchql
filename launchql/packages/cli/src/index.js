#!/usr/bin/env node
import { skitch } from './lib/cli';
var argv = require('minimist')(process.argv.slice(2));

// skitch upgrade
(async () => {
  await skitch(argv);
})();
