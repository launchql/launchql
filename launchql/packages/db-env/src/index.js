const envalid = require('envalid');
const { port, str } = envalid;

module.exports = envalid.cleanEnv(
  process.env,
  {
    PGPORT: port({ default: 5432 }),
    PGHOST: str({ default: 'localhost' }),
    PGUSER: str({ default: 'postgres' }),
    PGPASSWORD: str({ default: 'password' })
  },
  { dotEnvPath: null }
);
