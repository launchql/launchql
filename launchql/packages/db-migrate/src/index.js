export * from './export';
export * from './migrate';
