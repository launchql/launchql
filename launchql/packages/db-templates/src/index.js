
import * as column from './schemas/column';
import * as extension from './schemas/extension';
import * as fixture from './schemas/fixture';
import * as foreignKey from './schemas/foreignKey';
import * as fullPolicy from './schemas/fullPolicy';
import * as grantAllTables from './schemas/grantAllTables';
import * as grantExecute from './schemas/grantExecute';
import * as grantRole from './schemas/grantRole';
import * as grantSchema from './schemas/grantSchema';
import * as grantTable from './schemas/grantTable';
import * as importSchema from './schemas/importSchema';
import * as importTable from './schemas/importTable';
import * as index from './schemas/index';
import * as peoplestamps from './schemas/peoplestamps';
import * as policy from './schemas/policy';
import * as procedure from './schemas/procedure';
import * as role from './schemas/role';
import * as rowLevelSecurity from './schemas/rowLevelSecurity';
import * as schema from './schemas/schema';
import * as table from './schemas/table';
import * as timestamps from './schemas/timestamps';
import * as trigger from './schemas/trigger';
import * as type from './schemas/type';
import * as uniqueIndex from './schemas/uniqueIndex';
import * as utility from './schemas/utility';
import * as view from './schemas/view';
export default {
  column,extension,fixture,foreignKey,fullPolicy,grantAllTables,grantExecute,grantRole,grantSchema,grantTable,importSchema,importTable,index,peoplestamps,policy,procedure,role,rowLevelSecurity,schema,table,timestamps,trigger,type,uniqueIndex,utility,view
};