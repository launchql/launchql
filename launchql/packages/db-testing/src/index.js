export * from './lib/connection';
export * from './lib/db';
export * from './lib/utils';
export * from './lib/testing';
export * from './lib/sqitch';
export * from './lib/test-db';
export * from './lib/api';
export { wrapConn } from './lib/helpers';
