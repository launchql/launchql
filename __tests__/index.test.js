import * as ast from '../src';

it('works', () => {
  const obj = ast.A_Expr({
    kind: 0,
    name: [ast.String({ str: '=' })],
    lexpr: ast.Integer({ ival: 0 }),
    rexpr: ast.Integer({ ival: 0 })
  });
  expect(obj).toMatchSnapshot();
});
