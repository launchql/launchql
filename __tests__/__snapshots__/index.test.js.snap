// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
Object {
  "A_Expr": Object {
    "kind": 0,
    "lexpr": Object {
      "Integer": Object {
        "ival": 0,
      },
    },
    "name": Array [
      Object {
        "String": Object {
          "str": "=",
        },
      },
    ],
    "rexpr": Object {
      "Integer": Object {
        "ival": 0,
      },
    },
  },
}
`;
